Classes/Widgets/AbstractChartWidget.php

```diff
@@ -3,64 +3,66 @@
 
 namespace FriendsOfTYPO3\Dashboard\Widgets;
 
-abstract class AbstractChartWidget extends AbstractWidget
+use FriendsOfTYPO3\Dashboard\Widgets\Interfaces\AdditionalCssInterface;
+use FriendsOfTYPO3\Dashboard\Widgets\Interfaces\EventDataInterface;
+use FriendsOfTYPO3\Dashboard\Widgets\Interfaces\RequireJsModuleInterface;
+
+/**
+ * The AbstractChartWidget class is the basic widget class for all chart widgets
+ * Is it possible to extends this class for own widgets, but EXT:dashboard provide
+ * also some more special chart types of widgets. For more details, please check:
+ * @see AbstractBarChartWidget
+ * @see AbstractDoughnutChartWidget
+ * @see AbstractLineChartWidget
+ * More information can be found in the documentation.
+ * @TODO: Add link to documentation
+ */
+abstract class AbstractChartWidget extends AbstractWidget implements AdditionalCssInterface, RequireJsModuleInterface, EventDataInterface
 {
-    /**
-     * @var string
-     */
     protected $chartType = '';
 
-    /**
-     * @var array
-     */
     protected $chartData = [];
 
-    /**
-     * @var array
-     */
     protected $chartOptions = [];
 
-    /**
-     * @var array
-     */
+    protected $eventData = [];
+
     protected $chartColors = ['#f49702', '#a4276a', '#1a568f', '#4c7e3a', '#69bbb5'];
 
     protected $additionalClasses = 'dashboard-item--chart';
 
     protected $iconIdentifier = 'dashboard-chart';
 
     /**
-     * AbstractChartWidget constructor.
+     * Method to set all data for the chart in $this->eventData
      */
-    public function __construct()
-    {
-        parent::__construct();
-        $this->cssFiles[] = $this->publicResourcesPath . 'CSS/Dist/Chart.min.css';
-        $this->jsFiles['chartjs'] = $this->publicResourcesPath . 'JavaScript/Dist/Chart.min';
-        $this->jsFiles['chartinitializer'] = $this->publicResourcesPath . 'JavaScript/ChartInitializer';
-    }
+    abstract protected function prepareChartData(): void;
 
-    /**
-     * Method to set all data for the chart
-     */
-    protected function prepareChartData(): void
+    public function getEventData(): array
     {
-        // This method need to be implemented by the final class
+        $this->eventData['graphConfig'] = [
+            'type' => $this->chartType,
+            'data' => $this->chartData,
+            'options' => $this->chartOptions
+        ];
+        $this->prepareChartData();
```

Comment:
I guess this method should be called before the `graphConfig` is set. Now the `data` element is empty.

Reply:
correct, will push an update